### ISSUE TITLE
Documentation question: what's required to run etc1s/uastc from command line?

### ISSUE THREAD

**Issue by hybridherbst** [donmccurdy/glTF-Transform#134]
I followed the setup steps and installed the CLI as global npm package.

Running `gltf-transform etc1s input.glb output.glb` just prints an unhelpful
`error: Cannot read property 'forEach' of undefined`
and I'm not quite sure how to proceed.

Any help would be appreciated! Goal is to take a GLB and compress every texture inside as etc1s.

EDIT: Tried with another GLB file and this time getting a better error, asking me to install KTX-Software. Still weird that the first one didn't even get to that place.

**Comment by donmccurdy** [comment]
Hm, yeah it should be printing a hint to install KTX-Software when you use the `etc1s` or `uastc` commands. I'm not sure why that error would appear — is this on v0.7.5? Would you be able to share that first `input.glb`, or the results of `gltf-transform inspect input.glb`?

**Comment by hybridherbst** [comment]
The file certainly has validation errors regarding empty meshes (meshes with 0 vertices). I understand these aren't really allowed in GLB for whatever reason (they are certainly allowed in FBX).

I sent you the file in question via email.



Inspect produces the same error:

```
 INFO
 ────────────────────────────────────────────
┌────────────────────┬─────────────────────┐
│ generator          │                     │
├────────────────────┼─────────────────────┤
│ version            │ 2.0                 │
├────────────────────┼─────────────────────┤
│ extensionsUsed     │ KHR_lights_punctual │
├────────────────────┼─────────────────────┤
│ extensionsRequired │ none                │
└────────────────────┴─────────────────────┘

warn: Unable to parse document.

error: Cannot read property 'forEach' of undefined
```

**Comment by donmccurdy** [comment]
Thanks! I'll take a look. I can't think of a reason GLB wouldn't allow 0-vertex meshes, but perhaps I've missed an edge case in the implementation here.

**Comment by donmccurdy** [comment]
Fixed in v0.8.1 — you're right that a mesh must have at least one primitive, and a primitive must have at least one vertex in glTF. I guess the alternative would be to write a single point at `0,0,0` for the primitive, maybe? In either case, the files should read/write correctly with this library (even when primitives are omitted), but note that you'll get some incomplete results, from certain functions, for empty meshes. For example, the `bounds` function returns `Infinity, Infinity, Infinity`.
